### PR TITLE
Transform LWRPs on custom resources

### PR DESCRIPTION
**Pull request by Annih**
This is the new best practices.
This also fixes compatibility issues with chef-client >= 13 where there
is no more constants for resources in ::Chef::Resources.

### CODE REVIEW

File: resources/listener.rb [hunk 1]
```diff
@@ -18,27 +18,120 @@
...
+  data = registry_hash_values key_name
+  port data['Port']
+  enabled data['enabled'] == 1
+  hostname data['hostname']
+  url_prefix data['uriprefix']
+  certificate_thumbprint data['certThumbprint']
+end
+
+action :configure do
+  self_signed_certificate if https? && new_resource.certificate_thumbprint.to_s.empty?
```
Comment (jeremy-clerc) [inline resources/listener.rb]: I do not see any dep or chef resource providing `self_signed_certificate`, also it was not in the old format ?

File: resources/service_certmapping.rb [hunk 2]
```diff
@@ -17,24 +17,98 @@
...
+  begin
+    winrm_data = winrm_get
+  rescue StandardError => e
+    ::Chef::Log.warn("An error occured while retrieving certmapping: #{e.message}")
+  end
+  current_value_does_not_exist! if winrm_data.nil? || winrm_data.empty?
+
+  username winrm_data['UserName']
+  enabled(winrm_data['Enabled'] != 'false')
+end
```
Comment (jeremy-clerc) [inline resources/service_certmapping.rb:40]: We do not need to set `issuer`, `subject`, `uri` ? Like with `hydrate` previously ?

**Review by jeremy-clerc — CHANGES_REQUESTED** [review]
Just check my other two comments about the self_signed and load_value. Else it looks good.

File: resources/service_certmapping.rb (hunk 2 above)
Comment (Annih) [reply resources/service_certmapping.rb:40]: No we don't since these property are marked as `identity`

**Review by jeremy-clerc — APPROVED** [review]

**Pull request merged**